lib/ui/semantics.dart

```diff
@@ -786,6 +786,11 @@ abstract class SemanticsUpdateBuilder {
   /// z-direction starting at `elevation`. Basically, in the z-direction the
   /// node starts at `elevation` above the parent and ends at `elevation` +
   /// `thickness` above the parent.
+  ///
+  /// The `headingLevel` describes that this node is a heading, additionally
+  /// indicates the hierarchy level this node represents as a heading. A value
+  /// of -1 indicates that this node is not a heading. A value of 1 or greater
+  /// indicates that this node is a heading at the specified level.
```

Comment:
Should the parameter have an upper bound, and perhaps an `assert` that ensures that the upper bound is satisfied? It would also be useful to link to these pages:

- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-level

Another important thing to document is how it applies across platforms (seems like it's a noop outside web).

Reply:
I agree, it should have an upper bound, not sure if I put the assert in the correct place, can you please check it out?. I also added more details in the doc.